packages/bot-strategy-runner/src/LogCaptureTransport.ts

```diff
@@ -0,0 +1,32 @@
+// Custom winston transport to capture logs produced. Used by the strategy runner to get back execution outputs.
+
+import Transport = require("winston-transport");
+
+class logCaptureTransport extends Transport {
+  logStorage: any;
+  constructor(opts: any, logStorage: Array<any>) {
+    super(opts);
+    this.logStorage = logStorage;
+  }
+  async log(info: any, callback: any) {
+    try {
+      // eslint-disable-next-line prefer-const
+      let { timestamp, level, error, ...args } = info;
+      if (error) {
```

Comment:
is there a way the logger could not care about the data its logging?

Reply:
so the reason it needs to do this is that the error object is of type `Error` and this does not log nicely at all. If you want to store this within a json object you cant easily read it or consume it within a nested structure (such as within a GCP log). This code here basically strips out all nasty punctuation and enables this to be treated as a simple string.

I'm curious how you'd prefer this to be consumed. very open to any alternative pattern!